types/src/validator_set.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+use ccrypto::blake256;
+use ckey::Public;
+use primitives::H256;
+use rlp::{Decodable, DecoderError, Encodable, Rlp, RlpStream};
+
+// It will be hashed in the header.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct CompactValidatorSet(pub Vec<(Public, u64)>);
```

Comment:
Why do we need this structure?
Why now implementing hash function instead of create_compact_validator_set for NextValidator?

Reply:
I suggested to Junha to add this struct in the "types/src" directory.
If someone wants to know what is the "validator_set_root" in the header, reading this struct will help them.

Reply:
Because it will be also passed to the light client for verification. Light client only needs that compact data of the validator set.